core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala

```diff
@@ -98,12 +98,15 @@ trait RawlsApiService //(val workspaceServiceConstructor: UserInfo => WorkspaceS
 
   val baseApiRoutes = workspaceRoutes ~ entityRoutes ~ methodConfigRoutes ~ submissionRoutes ~ adminRoutes ~ userRoutes ~ billingRoutesV2 ~ billingRoutes ~ notificationsRoutes ~ servicePerimeterRoutes ~ snapshotRoutes
 
+  val instrumentedRoutes = instrumentRequest(baseApiRoutes)
+
   def apiRoutes =
     options(complete(OK)) ~
     withExecutionContext(ExecutionContext.global) { //Serve real work off the global EC to free up the dispatcher to run more routes, including status
-      baseApiRoutes
+      instrumentedRoutes
```

Comment:
poking into instrumentedRoutes a bit, I see that it calls mapResponse... noob question... will this work for error/exception cases as well as successful response cases?

Reply:
also, is there a unit test to assert the expected logging?

Reply:
> will this work for error/exception cases as well as successful response cases?
Short answer: 
It does work for both cases. 

Long answer:
A route can have 1 of 3 outcomes:
1. The route returns successfully (an httpResponse where status == 2XX) 
2. The route returns failure (an httpResponse where status == 4-5XX)
3. The route returns rejected (the route cannot process that request - and the next route gets a try at it)
If the route is 'sealed', #3 is mapped to #2, and the result is always guaranteed to be a completed http response. 
When run from a full application, so long as  the correct handlers are in scope, the routes are implicitly sealed.
So yes, this function gets the http response that is the result of the call, whether success or failure.

> also, is there a unit test to assert the expected logging?
Unit test.... eh.... not really...
Integrated checks.... there's quite a few that were already there.
I can write some new ones that are proper unit tests if we think it's worth the time to fight with the existing infrastructure (and ideally break up the existing tests into more digestible pieces).

Reply:
ok great.

Unit test question was more informational.  Not a hard blocker since the state of things sounds like a massive lift.   It wouldn't be fair to ask for such a heavy lift to make the overarching codebase more testable RIGHT now since the IRWG is actively exploring best practices and example of doing that :)